osx-utils/macos-bundler.py

```diff
@@ -0,0 +1,342 @@
+	revisionMatch = search(
+	pattern=r"-(?P<revisionPR>[0-9]+)-g?(?P<revisionCommit>[0-9a-fA-F]{7,})-?(?P<revisionType>(?:alpha|beta))?",string=package_version)
+	if revisionMatch != None:
+		revPR = revisionMatch.group('revisionPR')
+		revCommit = revisionMatch.group('revisionCommit')
+		revType = revisionMatch.group('revisionType') or None
+		if revType != None:
+			semVersion += (revType[0] + revPR)
+
+def clean_build_dir():
+	global performous_build_dir
+	print("--- Wiping temporary build folder: " + str(performous_build_dir))
+	shutil.rmtree(path=performous_build_dir, ignore_errors=True)
+	performous_build_dir.mkdir(mode=0o755, exist_ok=True)
+
+def create_dmg(fancy: bool = True):
+	outFile = (performous_out_dir / ("Performous-" + package_version + ".dmg"))
+	execute(fr"""
+		dmgbuild \
+		-s "{str(performous_source_dir / 'osx-utils/performous-dmg-settings.py')}" \
+		-Dapp="{str(performous_out_dir / 'Performous.app')}" \
+		-Dbackground="{str(performous_source_dir / 'osx-utils/resources/dmg-bg.png')}" \
+		-Dlicense="{str(performous_source_dir / 'LICENSE.md')}" \
+		"Performous-{package_version}" \
+		"{outFile}"
+	""")
+
+def bundle_libs():
+	global performous_out_dir
+	print("Copying dependencies and fixing linkage inside Performous.app...")
+	
+	execute(fr"""
+		dylibbundler -od -b \
+		-x "{performous_out_dir}/Performous.app/Contents/MacOS/Performous" \
+		-d "{performous_out_dir}/Performous.app/Contents/Resources/lib" \
+		-p @executable_path/../Resources/lib/
+	""")
+	return
+
+usageHelp = f"""\nPerformous macOS Bundler
+
+Usage:
+	macos_bundler.py [--arch <architecture>]... [--prefer-macports | --prefer-homebrew] [options]
+	macos_bundler.py [options]
+
+Options:
+	-h --help  Show this help message.
+	-b --preserve-build  Don't remove temporary build files before starting.
+	-d --debug  Create .app for debugging (skip copying and relinking of dependencies, as well as creation of .dmg package)
+	-f --flat-output  Put output directly in the output folder, without versioned folders.
+	-j <n>, --jobs <n>  Argument passed to make, specifying the max number of jobs to run. Defaults to the output of os.cpu_count() [default: {str(cpu_count())}]
+	-k --no-clean  Don't erase tree structure of the .app before beginning work.
+	-n --no-regenerate  Don't regenerate the CMake build system.
+	-v <version>, --package-version <version>  Sets the version on the Info.plist of the created .app. By default, it's calculated according to the latest stable version and hash of the latest commit.
+	-x --xcode-project  Generate an XCode project suitable for debugging. Note: to use this project, it's necessary to build the install target from XCode and then the performous target (or, alternatively, mirror the settings for the performous scheme on the install target)
+	--script-debug  Print the resolved arguments and options passed to this utility.
+
+Environment:
+	--arch <architecture>...  Target architecture names passed to the compiler. [default: x86_64] 
```

Comment:
would it be reasonable to make --arch's default the system architecture?

Reply:
Actually yes, come to think of it.